src/NuGet.Core/NuGet.LibraryModel/Library.cs

```diff
@@ -31,18 +45,25 @@ public class Library
 
         public override string ToString()
         {
-            if (Identity == null)
+            if (Identity is null)
             {
-                return LibraryRange?.ToString();
+                if (LibraryRange is null)
+                {
+                    return typeof(Library).FullName!;
```

Comment:
If `Identity` and `LibraryRange` can't be null (because they are required), I am curious to know why we need this if block in the first place.

Reply:
With all the nullable annotation PRs, I generally try to minimize changes, except to keep the compiler happy. So, I intentionally did not refactor old code, in order to avoid the PR doing too many things, and to save time (other projects took many, many hours to annotate and fix compiler errors).

Technically, marking properties as `required` will only break existing code that tries to compile, referencing the new assembly with the `required` attribute. The binaries are still ABI compatible, and therefore any binary that already creates instances of this class without setting a value, will have null values for these properties, despite it being marked as `required`. But as mentioned several times elsewhere, the likelyhood of anything other that NuGet.Packaging.dll creating instances of these classes is really low, so it's probably safe to change.

I can change it if the team prefers, but the above is an explanation of why I didn't do so earlier.